### PR TITLE
fix(versioning): use BigInt to avoid integer precision loss in gradle, maven and loose

### PR DESCRIPTION
**Pull request by RahulGautamSingh**
## Changes

Version tokenizers parsed numeric parts with `parseInt`/`Number`, losing precision above `Number.MAX_SAFE_INTEGER` (~9e15). Timestamp-style versions (e.g. `2026051723231779060202`) rounded to the same float, so `compare()` returned `0` ("equal") and no updates were detected.

This switches numeric tokens to `BigInt` in the `gradle`, `maven` and `loose` schemes so large versions compare with full precision. `getMajor`/`getMinor`/`getPatch` keep their `number | null` contract via `Number(...)`.

Reported in discussion https://github.com/renovatebot/renovate/discussions/43676.

## Context

- [ ] This closes an existing Issue, Closes: #
- [x] This doesn't close an Issue, but I accept the risk that this PR may be closed if maintainers disagree with its opening or implementation

## AI assistance disclosure

- [x] Yes — substantive assistance (AI-generated non‑trivial portions of code, tests, or documentation).

Code and tests written with Claude Code (Claude Opus 4.8).

## Documentation (please check one with an [x])

- [x] No documentation update is required

## How I've tested my work (please select one)

- [x] Newly added/modified unit tests

### CODE REVIEW

**Review by zharinov — CHANGES_REQUESTED** [review]
Just added some edge case handling, please check it out and merge if you agree:

- https://github.com/RahulGautamSingh/renovate/pull/36

File: lib/modules/versioning/maven/index.ts [hunk 1]
```diff
@@ -116,7 +120,42 @@ function isGreaterThan(a: string, b: string): boolean {
...
+  if (!(isVersion(a) && isVersion(b))) {
+    return false;
+  }
+
+  const left = getExactSection(a, type);
+  const right = getExactSection(b, type);
+  return left === right;
+}
+
+const isStable = (version: string): boolean => {
```
Comment (RahulGautamSingh) [inline lib/modules/versioning/maven/index.ts]:
```suggestion
function isStable(version: string): boolean {
```

File: lib/workers/repository/process/lookup/update-type.ts [hunk 2]
```diff
@@ -26,9 +26,15 @@ export function getUpdateType(
   ) {
     return 'major';
   }
+  const newMinor = versioningApi.getMinor(newVersion)!;
+  const currentMinor = versioningApi.getMinor(currentVersion)!;
```
Comment (viceice) [inline lib/workers/repository/process/lookup/update-type.ts:30]: we should add a comment to correctly handle null here later because I think this assumption is wrong

**Review by viceice — COMMENTED** [review]
otherwise LGTM

File: lib/workers/repository/process/lookup/update-type.ts (hunk 2 above)
Comment (RahulGautamSingh) [reply lib/workers/repository/process/lookup/update-type.ts]: added guard instead

**Review by viceice — APPROVED** [review]
